instrumentation/postgresql-simple/src/OpenTelemetry/Instrumentation/PostgresqlSimple.hs

```diff
@@ -105,254 +111,144 @@ staticConnectionAttributes Connection {connectionHandle} = liftIO $ do
---   option is ignored.
-foldWithOptions :: (MonadBracketError m, MonadLocalContext m, FromRow row, ToRow params)
-                => FoldOptions
-                -> Connection
-                -> Query
-                -> params
-                -> a
-                -> (a -> row -> m a)
-                -> m a
-foldWithOptions opts = _
-
--- | A version of 'foldWithOptions' taking a parser as an argument
-foldWithOptionsAndParser :: (MonadBracketError m, MonadLocalContext m, ToRow params)
-                         => FoldOptions
-                         -> Simple.RowParser row
-                         -> Connection
-                         -> Query
-                         -> params
-                         -> a
-                         -> (a -> row -> m a)
-                         -> m a
-foldWithOptionsAndParser opts parser conn template qs a f = _
-
--- | A version of 'fold' that does not perform query substitution.
-fold_ :: (MonadBracketError m, MonadLocalContext m, FromRow r) =>
-         Connection
-      -> Query                  -- ^ Query.
-      -> a                      -- ^ Initial state for result consumer.
-      -> (a -> r -> m a)       -- ^ Result consumer.
-      -> m a
-fold_ = _
-
--- | A version of 'fold_' taking a parser as an argument
-foldWith_ :: (MonadUnliftIO m, MonadBracketError m, MonadLocalContext m) =>
-             Simple.RowParser r
-          -> Connection
-          -> Query
-          -> a
-          -> (a -> r -> m a)
-          -> m a
-foldWith_ = _
-
-foldWithOptions_ :: (MonadUnliftIO m, MonadBracketError m, MonadLocalContext m, FromRow r) =>
-                    FoldOptions
-                 -> Connection
-                 -> Query             -- ^ Query.
-                 -> a                 -- ^ Initial state for result consumer.
-                 -> (a -> r -> m a)  -- ^ Result consumer.
-                 -> m a
-foldWithOptions_ opts conn query' a f = Simple.foldWithOptions_ opts conn query' a f
-
--- | A version of 'foldWithOptions_' taking a parser as an argument
-foldWithOptionsAndParser_ :: FoldOptions
-                          -> Simple.RowParser r
-                          -> Connection
-                          -> Query             -- ^ Query.
-                          -> a                 -- ^ Initial state for result consumer.
-                          -> (a -> r -> IO a)  -- ^ Result consumer.
-                          -> IO a
-foldWithOptionsAndParser_ opts parser conn query' a f = _
-
--- | A version of 'fold' that does not transform a state value.
-forEach :: (MonadUnliftIO m, MonadBracketError m, MonadLocalContext m, ToRow q, FromRow r) =>
-           Connection
-        -> Query                -- ^ Query template.
-        -> q                    -- ^ Query parameters.
-        -> (r -> m ())         -- ^ Result consumer.
-        -> m ()
-forEach = _
+
+pgsSpan :: Connection -> C.ByteString -> IO a -> IO a
```

Comment:
I think it would be nice to export some version of this in case future functions in postgresql-simple come along that someone wants to trace before we roll in support.

Also, I it'd be great to ensure that these wrapped functions use `HasCallStack` to ensure that the generated trace locations show the right function.

Reply:
Done, and done.